src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +39,646 @@ const ActorInit En_Rz_InitVars = {
+    }
+}
+
+void func_80BFC7E0(EnRz* this, PlayState* play) {
+    s32 new_var;
+
+    EnRz_UpdateSkelAnime(this, play);
+
+    new_var = this->timer;
+    if (new_var > 0) {
+        this->timer--;
+    } else {
+        this->actionFunc = func_80BFC8F8;
+        func_80BFB9E4(play, this, 2);
+    }
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC728;
+        func_80BFB9E4(play, this, 0);
+        this->actor.speedXZ = 0.0f;
+        func_80BFBDFC(play);
+    } else if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void func_80BFC8AC(EnRz* this, PlayState* play) {
+    this->timer = 100;
+    this->actionFunc = func_80BFC7E0;
+    this->actor.speedXZ = 0.0f;
+    func_80BFB9E4(play, this, 0);
+}
+
+void func_80BFC8F8(EnRz* this, PlayState* play) {
+    EnRz_UpdateSkelAnime(this, play);
+
+    this->actor.speedXZ = 1.5f;
+
+    switch (func_80BFBB44(this)) {
+        case 2:
+            func_80BFBA50(this, play);
+            func_80BFC8AC(this, play);
+            break;
+        case 1:
+            func_80BFC8AC(this, play);
+            break;
+    }
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actionFunc = func_80BFC728;
+        func_80BFB9E4(play, this, 0);
+        this->actor.speedXZ = 0.0f;
+        func_80BFBDFC(play);
+    } else if (EnRz_CanTalk(this, play)) {
+        func_800B8614(&this->actor, play, 120.0f);
+    }
+}
+
+void EnRz_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnRz* this = THIS;
+
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 40.0f, 25.0f, 40.0f, 5);
+
+    this->actionFunc(this, play);
+
+    if (DECR(this->unk_412) == 0) {
+        this->unk_412 = Rand_S16Offset(60, 60);
+    }
+
+    this->unk_410 = this->unk_412;
+    if (this->unk_410 >= 3) {
+        this->unk_410 = 0;
+    }
+}
+
+void EnRz_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnRz* this = THIS;
+
+    if (limbIndex == OBJECT_RZ_LIMB_0B) {
+        Matrix_MultVec3f(&D_80BFCD60, &thisx->focus.pos);
+    }
+    if (limbIndex == OBJECT_RZ_LIMB_03) {
+        Matrix_MultVec3f(&gZeroVec3f, &this->shadowPos);
+    }
+}
+
+void EnRz_Draw(Actor* thisx, PlayState* play) {
+    EnRz* this = THIS;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    func_8012C28C(play->state.gfxCtx);
+
+    if (!EN_RZ_GET_PARAM_8000(thisx)) {
+        AnimatedMat_DrawStepOpa(play, Lib_SegmentedToVirtual(&object_rz_Matanimheader_00D768), 0);
+    } else {
+        AnimatedMat_DrawStepOpa(play, Lib_SegmentedToVirtual(&object_rz_Matanimheader_00D768), 1);
+    }
+
+    if (this->animIndex == 4) {
+        gSPSegment(POLY_OPA_DISP++, 0x09, Lib_SegmentedToVirtual(D_80BFCCF0));
+    } else if (this->unk_420 & 2) {
```

Comment:
This should have defines for each used flag, the & 1 flag looks the same as this one in Ge2: https://github.com/EllipticEllipsis/mm/blob/7a189e81c9c11f36458dc9aacc9169f35c6bd1d3/src/overlays/actors/ovl_En_Ge2/z_en_ge2.c#L202-L225